Enigmatry.Entry.AspNetCore.Authorization/AuthorizationStartupExtensions.cs

```diff
@@ -0,0 +1,36 @@
+﻿using Enigmatry.Entry.AspNetCore.Authorization.Requirements;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Enigmatry.Entry.AspNetCore.Authorization;
+public static class AuthorizationStartupExtensions
+{
+    public static ControllerActionEndpointConventionBuilder AppRequireAuthorization(
+        this ControllerActionEndpointConventionBuilder builder,
+        bool enable) =>
+        !enable ? builder : builder.RequireAuthorization();
+
+    public static AuthorizationBuilder AppAddAuthorization(this IServiceCollection services, bool enable)
+    {
+        if (!enable)
```

Comment:
Don't see much sense in this flag. One can use AddAuthorizationBuilder directly without this extension?

Reply:
I agree with @zsrdjan on this one. Note that it is not technically mocking; we're just skipping the registration of the policies. IMHO, integration tests should focus on 1 integration, not as many as possible (which would make it a system test). So, it is good to have the flexibility to enable or disable authentication.
OTOH, if you do want to mock this, you could inject a dummy instance of the IAuthorizationProvider, that just returns true. But that does not add anything significant to your tests.